Lapka.Pets.Infrastructure/Queries/Handlers/GetPetsByRaceHandler.cs

```diff
@@ -0,0 +1,31 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Convey.CQRS.Queries;
+using Lapka.Pets.Application.Dto;
+using Lapka.Pets.Application.Queries;
+using Lapka.Pets.Application.Services;
+using Lapka.Pets.Core.Entities;
+
+namespace Lapka.Pets.Infrastructure.Queries.Handlers
+{
+    public class GetPetsByRaceHandler : IQueryHandler<GetPetsByRace, IEnumerable<PetBasicDto>>
+    {
+        private readonly IPetQueryService _petQueryService;
+
+        public GetPetsByRaceHandler(IPetQueryService petQueryService)
+        {
+            _petQueryService = petQueryService;
+        }
+        public async Task<IEnumerable<PetBasicDto>> HandleAsync(GetPetsByRace query)
+        {
+            if (string.IsNullOrEmpty(query.Race))
```

Comment:
If you dont allow empty value of string, throw exception, not returning empty collection